core/src/apps/management/recovery_device/homescreen.py

```diff
@@ -174,15 +204,42 @@
 
 async def _request_share_next_screen(ctx: wire.Context, mnemonic_type: int) -> None:
     if mnemonic_type == mnemonic.TYPE_SLIP39:
-        remaining = storage.recovery.get_remaining()
+        remaining = storage.recovery.fetch_slip39_remaining_shares()
+        group_count = storage.recovery.get_slip39_group_count()
         if not remaining:
             # 'remaining' should be stored at this point
             raise RuntimeError
-        if remaining == 1:
-            text = "1 more share"
+
+        if group_count > 1:
+            content = layout.RecoveryHomescreen(
+                "More shares needed", "for this recovery"
+            )
+            await layout.homescreen_dialog(ctx, content, "Enter share")
         else:
-            text = "%d more shares" % remaining
-        content = layout.RecoveryHomescreen(text, "needed to enter")
-        await layout.homescreen_dialog(ctx, content, "Enter share")
+            if remaining == 1:
+                text = "1 more share"
+            else:
+                text = "%d more shares" % remaining[0]
+            content = layout.RecoveryHomescreen(text, "needed to enter")
+            await layout.homescreen_dialog(ctx, content, "Enter share")
     else:
         raise RuntimeError
+
+
+async def _show_remaining_groups_and_shares(
+    ctx: wire.Context,
+    group_threshold: int,
+    mnemonics: List[str],
+    shares_remaining: List[int],
+) -> None:
+    identifiers = []
+    for m in mnemonics:
+        identifier = " ".join(m.split(" ")[0:3])
+        identifiers.append(identifier)
```

Comment:
I think we can use `set` right away, can't we? `set` does not raise an error on duplicates, it just ignores them.